e2e/nomostest/gitproviders/gitlab.go

```diff
@@ -68,190 +80,88 @@ func (g *GitlabClient) SyncURL(name string) string {
-func (g *GitlabClient) DeleteRepositories(names ...string) error {
-	var errs error
-
-	for _, name := range names {
-		id, err := GetProjectID(g, name)
-		if err != nil {
-			errs = multierr.Append(errs, fmt.Errorf("invalid repo name: %w", err))
-		} else {
-			out, err := exec.Command("curl", "-s", "--request", "DELETE",
-				fmt.Sprintf("https://gitlab.com/api/v4/projects/%s", id),
-				"--header", fmt.Sprintf("PRIVATE-TOKEN: %s", g.privateToken)).CombinedOutput()
-
-			if err != nil {
-				errs = multierr.Append(errs, fmt.Errorf("%s: %w", string(out), err))
-			}
-
-			response := string(out)
-			// Check for successful deletion (202 Accepted)
-			if strings.Contains(response, "\"message\":\"202 Accepted\"") {
-				continue
-			}
-			// Check if project is already marked for deletion (this is also a success case)
-			if strings.Contains(response, "Project has already been marked for deletion") {
-				continue
-			}
-			// Any other response is treated as an error
-			return errors.New(response)
-		}
-	}
-	return errs
+// DeleteRepositories is a no-op because Gitlab repo names are determined by the
+// test cluster name and RSync namespace and name, so they can be reset and reused
+// across test runs
+func (g *GitlabClient) DeleteRepositories(_ ...string) error {
+	g.logger.Info("[Gitlab] Skip deletion of repos")
+	return nil
 }
 
-// DeleteObsoleteRepos deletes all projects that has been inactive more than 24 hours
+// DeleteObsoleteRepos is a no-op because Gitlab repo names are determined by the
+// test cluster name and RSync namespace and name, so it can be reused if it
+// failed to be deleted after the test.
 func (g *GitlabClient) DeleteObsoleteRepos() error {
-	repos, _ := g.GetObsoleteRepos()
-
-	err := g.DeleteRepoByID(repos...)
-	return err
+	return nil
 }
 
-// DeleteRepoByID calls the DELETE API to delete the list of project id in Gitlab.
-func (g *GitlabClient) DeleteRepoByID(ids ...string) error {
-	var errs error
-
-	for _, id := range ids {
-		out, err := exec.Command("curl", "-s", "--request", "DELETE",
-			fmt.Sprintf("https://gitlab.com/api/v4/projects/%s", id),
-			"--header", fmt.Sprintf("PRIVATE-TOKEN: %s", g.privateToken)).CombinedOutput()
-
-		if err != nil {
-			errs = multierr.Append(errs, fmt.Errorf("%s: %w", string(out), err))
-		}
-
-		response := string(out)
-		// Check for successful deletion (202 Accepted)
-		if strings.Contains(response, "\"message\":\"202 Accepted\"") {
-			continue
-		}
-		// Check if project is already marked for deletion (this is also a success case)
-		if strings.Contains(response, "Project has already been marked for deletion") {
-			continue
-		}
-		// Any other response is treated as an error
-		return fmt.Errorf("unexpected response in DeleteRepoByID: %s", response)
+// sendRequest sends an HTTP request to the Gitlab API.
+func (g *GitlabClient) sendRequest(method, url string, body io.Reader) (*http.Response, error) {
```

Comment:
This is a good helper function. For even better request lifecycle management, consider adding a context.Context to sendRequest (and its callers like CreateRepository). You could then use http.NewRequestWithContext to allow for request cancellation and timeouts.